lib/compiler.js

```diff
@@ -60,19 +58,14 @@ function compile(code, options) {
     });
 
     importExportVisitor.finalizeHoisting();
+    result.identical = false;
+  }
 
-    if (! getOption(options, "repl")) {
-      result.code =
-        "module.run(function(){" +
-        magicString.toString() +
-        "\n})";
```

Comment:
The problem with removing this functionality from the compiler is that tools like Webpack or Meteor that use the compiler directly won't get the `module.run` wrapping, and may not be able to decide whether they should do it themselves, because they don't have access to the same information as the compiler.